### PR TITLE
Stop blockchain

### PR DESCRIPTION
**Pull request by j-a-m-l**
## Proposed changes

The intention of this changes is creating a way to stop the blockchain. It'd be used on tests to:
 1. Reduce the moving parts: running the blockchain on some tests produces, sometimes, side effects. This could be used to investigate them.
 2. Increase the speed. Stopping and re-starting the blockchain would be faster than creating it on every test.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [x] Test (adding missing tests or fixing existing tests)

I've moved the commit from https://github.com/ArkEcosystem/core/pull/783, to split that PR into smaller autonomous things

### CODE REVIEW

File: packages/core-blockchain/lib/index.js [hunk 1]
```diff
@@ -11,12 +12,15 @@ exports.plugin = {
...
+    blockchain = new Blockchain(options.networkStart)
 
     if (!process.env.ARK_SKIP_BLOCKCHAIN) {
       await blockchain.start()
     }
 
     return blockchain
+  },
+  async deregister (container, options) {
+    await blockchain.stop()
```
Comment (faustbrian) [inline packages/core-blockchain/lib/index.js]: No need for that wonky variable solution, can simply do `await container.resolve('blockchain').stop()`.

Comment (j-a-m-l) [reply packages/core-blockchain/lib/index.js]: Thanks @faustbrian, I've updated it with your suggestion

**Comment by faustbrian** [comment]
Could you run the tests locally? Travis has a lot of timeout issues the last few days and everytime I run the blockchain tests some random one fails.

**Comment by j-a-m-l** [comment]
I did it and 13 tests we're failing in both cases.

**Comment by j-a-m-l** [comment]
All tests are OK

**Comment by alexbarnsley** [comment]
@j-a-m-l I did remove this comment but re-adding. It's not major but I think it looks better with your STOP fix, just doesn't seem to get triggered during a sync for me

>i just re-tested this. Looks like it doesn't stop during a sync on devnet:
>
>My local forging node (in sync):
>image
>
>My devnet node (sync in progress):
>image
>
>The `The blockchain has been stopped` message doesn't show during

**Pull request merged**